### PR TITLE
fix: Enforce voting threshold param in oracle endblocker

### PR DESCRIPTION
**Pull request by rbajollari**
## Description

When updating the oracle rewarding in Ojo we noticed that the param `VoteThreshold` was not being used to enforce a minimum percentage of votes for a ballot to pass. This PR adds that enforcement and a unit test for it. 

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/oracle/abci.go [hunk 1]
```diff
@@ -46,6 +46,10 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) error {
 
 		// Iterate through ballots and update exchange rates; drop if not enough votes have been achieved.
 		for _, ballotDenom := range ballotDenomSlice {
+			if sdk.NewDecWithPrec(ballotDenom.Ballot.Power(), 2).LTE(k.VoteThreshold(ctx)) {
```
Comment (robert-zaremba) [inline x/oracle/abci.go:49]: Why the precision==2? Could you add a comment in the code?

File: x/oracle/abci.go [hunk 2]
```diff
@@ -46,6 +46,10 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) error {
...
+				continue
+			}
```
Comment (robert-zaremba) [inline x/oracle/abci.go:51]: We should add a warning log if the price is not updated.

**Review by robert-zaremba — COMMENTED** [review]
So we are ignoring an update and will have updated price? We need some notification about that.

File: x/oracle/abci.go [hunk 3]
```diff
@@ -57,7 +57,14 @@ func CalcPrices(ctx sdk.Context, params types.Params, k keeper.Keeper) error {
 
 	// Iterate through ballots and update exchange rates; drop if not enough votes have been achieved.
 	for _, ballotDenom := range ballotDenomSlice {
+		// Convert ballot power to a percentage to compare with VoteThreshold param
+		if sdk.NewDecWithPrec(ballotDenom.Ballot.Power(), 2).LTE(k.VoteThreshold(ctx)) {
+			ctx.Logger().Info("Ballot voting power is under vote threshold, dropping ballot for", ballotDenom)
```
Comment (adamewozniak) [inline x/oracle/abci.go:62]: I believe the last two args here are supposed to act like a key value pair
```suggestion
			ctx.Logger().Info("Ballot voting power is under vote threshold, dropping ballot", "denom", ballotDenom)
```

File: x/oracle/abci.go [hunk 4]
```diff
@@ -57,7 +57,14 @@ func CalcPrices(ctx sdk.Context, params types.Params, k keeper.Keeper) error {
...
+			continue
+		}
+
 		denom := strings.ToUpper(ballotDenom.Denom)
+
```
Comment (adamewozniak) [inline x/oracle/abci.go:67]: is this a lint suggestion?
```suggestion
```

File: x/oracle/abci_test.go [hunk 5]
```diff
@@ -22,8 +23,10 @@ import (
 )
 
 const (
-	displayDenom string = appparams.DisplayDenom
-	bondDenom    string = appparams.BondDenom
+	displayDenom     string = appparams.DisplayDenom
+	bondDenom        string = appparams.BondDenom
+	preVoteBlockDiff int64  = 2
+	voteBlockDiff    int64  = 3
```
Comment (adamewozniak) [inline x/oracle/abci_test.go:29]: would these change if parameters change? 

if so, we need to derive them from the parameters

**Review by adamewozniak — APPROVED** [review]
pre-approving with a couple comments 👍

File: x/oracle/abci_test.go (hunk 5 above)
Comment (rbajollari) [reply x/oracle/abci_test.go]: done👍

File: x/oracle/abci.go (hunk 4 above)
Comment (rbajollari) [reply x/oracle/abci.go]: No. I took it out.

**Comment by toteki** [comment]
@mergifyio backport release.v4.1.x

**Comment by toteki** [comment]
https://github.com/Mergifyio backport release/v4.1.x

**Pull request merged**